api/app/models/resource.rb

```diff
@@ -90,6 +94,12 @@ def resource_to_event
     )
   end
 
+  def queue_fetch_thumbnail
+    return unless external_id_changed? || new_record?
+    return if variant_thumbnail.present? && !external_id_changed?
```

Comment:
Another thought... logic is in the wrong place here. The ThumbnailFetcher model is responsible for fetching, so the decision as to whether or not it can fetch a thumbnail should be up to it, not up to the resource. 

Look at this approach below, in which we delegate each part of the decision to (in my view) the correct model. We leave it to the fetcher to determine which kinds of resources it can handle. We leave it to the resource to decide whether it should request a fetch based on its current state. Normally, I wouldn't include the Thumbnail::Fetcher.accepts? call below, and would instead just pass the resource to the fetcher and have it do nothing if it chose not to attempt a fetch — but in this case, there's some overhead involved in creating the redis job, so I think we should avoid it if it's unnecessary. 

In the resource:
```
after_commit :queue_fetch_thumbnail

def fetch_thumbnail?
  return unless Thumbnail::Fetcher.accepts?(self)
  external_id_changed? || !variant_thumbnail.present?
end

def queue_fetch_thumbnail
  return unless :fetch_thumbnail?
  FetchVideoThumbnails.perform_later(id)
end
```

In Thumbnail::Fetcher:
```
def self.accepts?(resource)
  return resource.is_external_video?
end

def fetch(resource)
  return unless self.class.accepts?(resource)
  # Check if an attempt has been made and failed (and that nothing has changed to warrant another attempt)
  return unless already_attempted?(resource) 
  [...]
end
```